spec/lib/publishing_api_payload_spec.rb

```diff
@@ -18,6 +18,62 @@
       expect(payload).not_to include(:first_published_at)
     end
 
+    it "includes a public_updated_at timestamp for draft editions" do
+      freeze_time do
+        edition = build(:edition)
+
+        payload = described_class.new(edition).payload
+
+        payload_hash = {
+          public_updated_at: Time.current,
+        }
+
+        expect(payload).to match a_hash_including(payload_hash)
+      end
+    end
+
+    it "includes a first_published_at and public_updated_at timestamp for published editions" do
+      edition = build(:edition, :published, first_published_at: "2020-02-20 08:00:00")
+
+      payload = described_class.new(edition).payload
+
+      payload_hash = {
+        first_published_at: Time.zone.parse("2020-02-20 08:00:00"),
+        public_updated_at: Time.zone.parse("2020-02-20 08:00:00"),
+      }
+
+      expect(payload).to match a_hash_including(payload_hash)
+    end
+
+    it "includes the full change history for published edition in reverse chronological order" do
+      change_history = [
+        {
+          id: SecureRandom.uuid,
+          note: "A major edit",
+          public_timestamp: "2020-02-20 09:30:00",
+        },
+      ]
+      first_edition = build(:edition, :published, first_published_at: "2020-02-10 09:30:00")
+      second_edition = build(:edition, :published, document: first_edition.document, change_history: change_history)
+
+      payload = described_class.new(second_edition).payload
+
+      expected_details_hash = {
+        change_history: [
+          {
+            note: "A major edit",
+            public_timestamp: Time.zone.parse("2020-02-20 09:30:00"),
+          },
+          {
+            note: "First published.",
+            public_timestamp: Time.zone.parse("2020-02-10 09:30:00"),
+          },
+        ],
+      }
+
+      expect(payload[:details]).to match a_hash_including(expected_details_hash)
```

Comment:
I don't think you need to test these bits here since we're covering a bunch of this in the history object test.

We can probably do something like:

```
it "delegates to PublishingApiPayload::History to populate change_history" do
  history = instance_double(
    PublishingApiPayload::History, 
    change_history: [{ note: "note", public_timestamp: Time.current }],
    public_updated_at: Time.current,
    first_published_at: Time.current,
  )
  allow(PublishingApiPayload::History).to receive(:new).and_return(history)
  payload = described_class.new(build(:edition)).payload
  expect(payload[:details][:change_history]).to match(history.change_history)
end
```

Reply:
Sure.  I've replaced the test with the one you've proposed, and ensured the `PublishingApiPayload::History` spec covers the situations we were previously testing here.